lib/config.rb

```diff
@@ -6,7 +6,18 @@
 require 'config/sources/hash_source'
 require 'config/sources/env_source'
 require 'config/validation/schema'
-require 'deep_merge/core'
+
+begin
+  gem 'deep_merge', '~> 1.2', '>= 1.2.1'
+  require 'deep_merge/core'
+rescue LoadError
```

Comment:
I think here like @pkuczynski said, it is better to raise an error if deep_merge is not available in a non Rails codebase. Since this gem is can be used in any ruby codebase, it is better to warn for Rails and raise error for non rails app if it is not available,